File: app/src/main/java/com/romankaranchuk/musicplayer/data/db/MusicRepository.java
package com.romankaranchuk.musicplayer.data.db;

import androidx.annotation.NonNull;

import com.romankaranchuk.musicplayer.data.Album;
import com.romankaranchuk.musicplayer.data.Song;

import java.util.List;

public interface MusicRepository {

    boolean saveAlbum(@NonNull Album album, @NonNull List<Song> songs);

    void saveSongs(@NonNull List<Song> songs);

    void deleteAlbum(@NonNull Album album);

    void deleteSong(@NonNull Song song);

    @NonNull
    List<Album> getAlbums();

    @NonNull
    List<Song> getSongs(@NonNull String albumId, boolean sortByName);
}

File: app/src/main/java/com/romankaranchuk/musicplayer/data/db/MusicRepository.kt
package com.romankaranchuk.musicplayer.data.db

import com.romankaranchuk.musicplayer.data.Album
import com.romankaranchuk.musicplayer.data.Song

interface MusicRepository {
    fun saveAlbum(album: Album, songs: List<Song>)
    fun saveSongs(songs: List<Song>)
    fun deleteAlbum(album: Album)
    fun deleteSong(song: Song)
    val albums: List<Album>
    fun getSongs(albumId: String, sortByName: Boolean): List<Song>
    fun getSong(songId: String): Song
}